File: src/main/kotlin/com/isel/ps/secdash/model/repositories/ExternalRepository.kt
package com.isel.ps.secdash.model.repositories

import com.isel.ps.secdash.model.Platform
import java.time.Instant

//import java.util.Date

data class ExternalRepository(
    val name: String,
    val externalId: String,
    val platform: Platform,
    val externalOwner: ExternalOwner,
    val htmlUrl: String,
    val description: String?,
    val issuesCount: Int,
    val createdAt: Instant,
    val updatedAt: Instant,
    val forksCount: Int,
    val visibility: Repository.Visibility,
)


data class ExternalOwner(
    val externalId: String,
    val name : String,
    val url : String,
    val avatarUrl: String?,
    val platform: Platform
)

File: frontend/src/model/repository/externalRepository.ts
import type { Platform, Visibility } from "./repository"

export type ExternalOwner = {
    externalId: string,
    name: string,
    url: string,
    avatarUrl: string | null,
    platform: Platform,
}

export type ExternalRepository = {
    name: string,
    externalId: string,
    platform: Platform,
    externalOwner: ExternalOwner,
    htmlUrl: string,
    description: string | null,
    issuesCount: number,
    createdAt: string,
    updatedAt: string,
    forksCount: number,
    visibility: Visibility,
}
